src/test/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureServiceUnitSpec.scala

```diff
@@ -0,0 +1,225 @@
+        val testObjectId = ManagedIdentityObjectId(UUID.randomUUID().toString)
+        val testActionManagedIdentity = ActionManagedIdentity(testActionManagedIdentityId, testObjectId, testDisplayName, testMrgCoordinates)
+
+        when(mockDirectoryDAO.loadActionManagedIdentity(testActionManagedIdentityId, samRequestContext)).thenReturn(IO.pure(Option(testActionManagedIdentity)))
+
+        // Act
+        val (ami, created) =
+          azureService.getOrCreateActionManagedIdentity(testResource, testAction, testBillingProfileId, samRequestContext).unsafeRunSync()
+
+        // Assert
+        ami should be(testActionManagedIdentity)
+        created should be(false)
+        verify(mockCrlService, never).buildMsiManager(testMrgCoordinates.tenantId, testMrgCoordinates.subscriptionId)
+      }
+
+      "delete an existing Action Managed Identity" in {
+        // Arrange
+        val mockCrlService = mock[CrlService]
+        val mockDirectoryDAO = mock[DirectoryDAO]
+        val mockAzureManagedResourceGroupDAO = mock[AzureManagedResourceGroupDAO]
+        val mockMsiManager = mock[MsiManager]
+        val mockIdentities = mock[Identities]
+
+        val azureService = new AzureService(mockCrlService, mockDirectoryDAO, mockAzureManagedResourceGroupDAO)
+
+        val testMrgCoordinates = ManagedResourceGroupCoordinates(
+          TenantId(UUID.randomUUID().toString),
+          SubscriptionId(UUID.randomUUID().toString),
+          ManagedResourceGroupName(UUID.randomUUID().toString)
+        )
+        val testBillingProfileId = BillingProfileId(UUID.randomUUID().toString)
+        val testAction = ResourceAction("testAction")
+        val testResource = FullyQualifiedResourceId(ResourceTypeName("testResourceType"), ResourceId("testResource"))
+        val testActionManagedIdentityId = ActionManagedIdentityId(testResource, testAction, testBillingProfileId)
+        val testDisplayName = azureService.toManagedIdentityNameFromAmiId(testActionManagedIdentityId)
+        val testObjectId = ManagedIdentityObjectId(UUID.randomUUID().toString)
+        val testActionManagedIdentity = ActionManagedIdentity(testActionManagedIdentityId, testObjectId, testDisplayName, testMrgCoordinates)
+
+        when(mockDirectoryDAO.loadActionManagedIdentity(testActionManagedIdentityId, samRequestContext)).thenReturn(IO.pure(Option(testActionManagedIdentity)))
+        when(mockCrlService.buildMsiManager(testMrgCoordinates.tenantId, testMrgCoordinates.subscriptionId)).thenReturn(IO.pure(mockMsiManager))
+        when(mockMsiManager.identities()).thenReturn(mockIdentities)
+        doNothing.when(mockIdentities).deleteById(testObjectId.value)
+        when(mockDirectoryDAO.deleteActionManagedIdentity(testActionManagedIdentityId, samRequestContext)).thenReturn(IO.pure(()))
+
+        // Act & Assert
+        azureService.deleteActionManagedIdentity(testActionManagedIdentityId, samRequestContext).unsafeRunSync()
```

Comment:
no assertions? i guess if it completes without error we are good?

Reply:
Yup! If one of the mocked functions wasn't called, ScalaTest will complain